harness/determined/keras/callbacks.py

```diff
@@ -0,0 +1,27 @@
+from typing import Any, Dict
+
+import tensorflow as tf
+
+
+class Callback(tf.keras.callbacks.Callback):  # type: ignore
+    """
+    A Determined implementation of Callback which supports additional new callbacks.
+    """
+
+    def on_validation_period_begin(self, complete_periods: int, logs: Dict) -> None:
+        pass
+
+    def on_validation_period_end(self, complete_periods: int, logs: Dict) -> None:
+        pass
+
+    def on_train_step_begin(self, total_batches_trained: int, logs: Dict) -> None:
+        pass
+
+    def on_train_step_end(self, total_batches_trained: int, logs: Dict) -> None:
+        pass
+
+    def get_state(self) -> Any:
```

Comment:
non-blocking: rename to `state_dict()` to match to `PytorchTrial.py`

Reply:
I don't think having the same name as pytorch is a valuable reason here.

I also think it's weird to require that this is a dictionary, which is why I removed the `_dict` from the naming I had in the rfc.